ert3/algorithms/_sensitivity.py

```diff
@@ -41,3 +45,148 @@ def one_at_the_time(
+            omega_temp[idx] = omega[1:]
+            k = range(i * sample_size, (i + 1) * sample_size)
+
+            # Saltelli et al. 1999
+            # https://www.tandfonline.com/doi/abs/10.1080/00401706.1999.10485594
+            phi = 2 * pi * np.random.rand()
+
+            # Xu and Gertner 2007:
+            # https://www.sciencedirect.com/science/article/pii/S0167947307001582
+            # https://github.com/cran/sensitivity/blob/master/R/fast99.R
+            if dist.type == "uniform":
+                for j in range(param_size):
+                    g = 0.5 + (1 / pi) * np.arcsin(np.sin(omega_temp[j] * s + phi))
+                    samples[k, j] = scipy.stats.uniform.ppf(
+                        g,
+                        loc=dist.lower_bound,
+                        scale=dist.upper_bound - dist.lower_bound,
+                    )
+            else:
+                for j in range(param_size):
+                    g = 0.5 + (1 / pi) * np.arcsin(np.sin(omega_temp[j] * s + phi))
+                    samples[k, j] = scipy.stats.norm.ppf(
+                        g, loc=dist.mean, scale=dist.std
+                    )
+
+        records = []
+        for sample in samples:
+            data = {idx: val for idx, val in zip(dist.index, sample)}
+            record = NumericalRecord(data=data, index=dist.index)
+            records.append(record)
+        group_records.append(records)
+
+    assert len(group_records) == len(parameters)
+    evaluations = []
+    for i in zip(*group_records):
+        evaluation = {group: record for group, record in zip(parameters.keys(), i)}
+        evaluations.append(evaluation)
+
+    return evaluations
+
+
+def fast_analyze(
+    parameters: MutableMapping[str, Distribution],
+    model_output: MultiEnsembleRecord,
+    harmonics: Optional[int],
+) -> Dict[Any, Dict[str, float]]:
+    # TODO: deal with model output optionals, will remove type: ignore
+    if len(parameters) == 0:
+        raise ValueError("Cannot study the sensitivity of no variables")
+    if model_output.record_names and len(model_output.record_names) > 1:
+        raise ValueError("Cannot analyze sensitivity with multiple outputs")
+    if model_output.record_names and len(model_output.record_names) < 1:
+        raise ValueError("Cannot analyze sensitivity with no output")
+
+    if harmonics is None:
+        harmonics = 4
+
+    param_indices = []
+    param_size = 0
+    for _, dist in parameters.items():
+        param_size += dist._size
+        for i in dist.index:
+            param_indices.append(i)
+
+    if model_output.ensemble_size % param_size == 0:
+        sample_size = int(model_output.ensemble_size / param_size)
+    else:
+        raise ValueError(
+            "The size of the model output must be a multiple of the number of parameters"
+        )
+
+    omega = floor((sample_size - 1) / (2 * harmonics))
+
+    data = []
+    record_name = model_output.record_names[0]  # type: ignore
```

Comment:
I agree. That would also allow for analyzing several outputs. However, the parameters contributing to each output would have to be specified as well.
This would be the case for other algorithms as well. I don't know if it should be added at this point when the algorithm configuration is specified in `ExperimentConfig`.

Reply:
I agree, lets leave that for later :+1: